### PR TITLE
Runs pages-layout tests aginst the local container using Firefox

### LINKED ISSUE #3656: [functional testing]  pages-layout tests need update for tor based testing
Part of epic #3659.

## Description

As we are moving towards Tor browser based functional tests, the pages-layout tests also need update to work properly.

Right now all of them happily failing on the `local_tests` branch.

### PR DESCRIPTION
**Pull request by kushaldas**
## Status

Ready for review

## Description of Changes

Fixes #3656

We now can run the `pages-layout` tests against the local container.

## Testing

```
0. cd securedrop
1. ./bin/dev-shell ./bin/run-test -v tests/pages-layout/
```

To run the full test suite

```
$ ./bin/dev-shell ./bin/run-test -v tests
```

### If you made changes to the server application code:

- [ ] Linting (`make ci-lint`) and tests (`make -C securedrop test`) pass in the development container

### If you made changes to `securedrop-admin`:

- [ ] Linting and tests (`make -C admin test`) pass in the admin development container

### If you made changes to the system configuration:

- [ ] [Configuration tests](https://docs.securedrop.org/en/latest/development/testing_configuration_tests.html) pass

### If you made non-trivial code changes:

- [ ] I have written a test plan and validated it for this PR

### If you made changes to documentation:

- [ ] Doc linting (`make docs-lint`) passed locally

### CODE REVIEW

**Comment by conorsch** [comment]
I'm getting a _single_ test failure across the whole suite:

```
1 failed, 458 passed, 2 xfailed in 6050.89 seconds

tests/pages-layout/test_journalist.py::TestJournalistLayout::test_delete_all[en_US] FAILED 
```

Will try cleaning out my local env and re-running. Reporting in case anyone else comes across same.

**Comment by kushaldas** [comment]
@conorsch Can you please rerun the whole test suite a few times? and also please paste the failed error details. I am wondering why did it fail :(

**Comment by redshiftzero** [comment]
Just ran the pages-layout tests on this branch, all passing, no issues here. This is good to merge from my end. 

@conorsch can you re-run that one failed test and see if it's a flake?

**Comment by conorsch** [comment]
@redshiftzero On the one hand, pleased to report that on subsequent runs, I could not reproduce that particular failure. Does sound like a flake! On the other hand, after several runs, I've yet to encounter 100% passing yet. Since the test suite takes a while (~1hr) to run, don't have too much data to report yet. I'll kick off another few builds and report back if anything is regular. Will make sure to leave some tests running while I'm not also working on other things, in case it's load-related (plausible if timeouts come into play).

**Comment by conorsch** [comment]
Successful test run! :tada:

```
459 passed, 2 xfailed in 6030.73 seconds
```

Will run the whole suite a few more times to try to shake out any problematic tests, but performance is good so far. Haven't yet taken a hard look at the diff presented here, but the commit messages are clearly written. We should be good for merge here shortly!

File: securedrop/tests/functional/journalist_navigation_steps.py [hunk 1]
```diff
@@ -17,6 +18,9 @@ def get_journalist_usernames():
     yield "dellsberg"
     yield "jpb"
     yield "bassel"
+    while True:
+        num = random.randint(1000,1000000)
+        yield "journalist" + str(num)
```
Comment (conorsch) [inline securedrop/tests/functional/journalist_navigation_steps.py]: @kushaldas Can you add a comment stating _why_ we're doing this? The commit message indicates that we _can_, but is unclear in terms of the motivation. Also:

* the comment above the function is now quite inaccurate (states that we're adding _two_ usernames :slightly_smiling_face:)
* flake8 violations in the diff presented

There are a few flake8 violations in the test files, but we can clean up the test prior to merge into develop.

**Comment by conorsch** [comment]
Nits addressed! Tests have been fairly stable for me. Occasionally I get a single failing test, but it's absolutely a flake—mostly it happens when my machine is under a lot of load, so perhaps it's related to the time.sleeps (#3486).

**Pull request merged**